contracts/interchain-token-service/src/primitives.rs

```diff
@@ -110,7 +145,9 @@ impl Message {
     pub fn token_id(&self) -> TokenId {
         match self {
             Message::InterchainTransfer(InterchainTransfer { token_id, .. })
-            | Message::DeployInterchainToken(DeployInterchainToken { token_id, .. }) => *token_id,
+            | Message::DeployInterchainToken(DeployInterchainToken { token_id, .. })
+            | Message::LinkToken(LinkToken { token_id, .. }) => *token_id,
+            Message::RegisterToken(_) => panic!("no token id associated with this message type"), // TODO is this right?
```

Comment:
I don't like circumventing the type system here. Either we need to treat RegisterToken separately or token_id() should return an option